lib/msprime.c

```diff
@@ -431,6 +433,11 @@ msp_set_migration_matrix(msp_t *self, size_t size, double *migration_matrix)
                     goto out;
                 }
             }
+            sum += migration_matrix[j * N + k];
+        }
+        if (self->model.type == MSP_MODEL_DTWF && sum > 1) {
+            ret = MSP_ERR_DTWF_MIGRATION_MATRIX_NOT_STOCHASTIC;
+            goto out;
```

Comment:
Hm - won't this be caught by the other check below, anyways? If so, is there a reason to keep it in both places? It seems like the code would be tidier with just one check...

Reply:
I agree it'd be better if we just had one check, but maybe there's some loophole we're not seeing?

Reply:
Don't know. I guess it doesn't hurt to have it in here, and arguably makes the constraint more obvious.

Reply:
If we can't figure out a way of testing it, though, we should definitely remove it - no point in having unreachable code cluttering the place up.

Reply:
I agree, this really isn't needed if checking on the fly, I will go ahead and amend.

Actually I think the check that's not covered by unit tests is the one in `msp_run_dtwf` ... this one is explicitly tested. I'll add that.

Thanks!